Update current release (#120)
trento-project/docs: trento/adoc/trento-kubernetes-install.adoc

include::product-attributes.adoc[]

[[sec-trento-k8s-deployment]]
=== {k8s} deployment
:revdate: 2025-10-16

The subsection uses the following placeholders:

* `TRENTO_SERVER_HOSTNAME`: the host name used by the end user to access the console.
* `ADMIN_PASSWORD`: the password of the default user created during the installation process. It must have at least 8 characters.

[[sec-trento-install-trentoserver-on-existing-k8s-cluster]]
==== Installing {trserver} on an existing {k8s} cluster

> {trserver} consists of a several components delivered as container images and intended for deployment on a {k8s} cluster. A manual production-ready deployment of these components requires {k8s} knowledge. Customers without in-house {k8s} expertise and who want to try {trento} with a minimum of effort, can use the {trento} Helm chart. This approach automates the deployment of all the required components on a single {k8s} cluster node. You can use the {trento} Helm chart to install {trserver} on a existing {k8s} cluster as follows:

Reviewer: ```suggestion
{trserver} consists of a several components delivered as container images and intended for deployment on a {k8s} cluster. A manual production-ready deployment of these components requires {k8s} knowledge. Customers without in-house {k8s} expertise and those who want to try {trento} with a minimum of effort, can use the {trento} Helm chart. This approach automates the deployment of all the required components on a single {k8s} cluster node. You can use the {trento} Helm chart to install {trserver} on a existing {k8s} cluster as follows:
```

. Install Helm:
+
====
[source,bash,subs="attributes"]
----
curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
----
====
+
. Connect Helm to an existing {k8s} cluster.
. Use Helm to install {trserver} with the {trento} Helm chart:
+
====
[source,bash,subs="attributes"]
----
helm upgrade \
   --install trento-server oci://registry.suse.com/trento/trento-server \
   --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \
   --set trento-web.adminUser.password=ADMIN_PASSWORD
----
====
+
When using a Helm version lower than 3.8.0, an experimental flag must be set as follows:
+
====
[source,bash]
----
HELM_EXPERIMENTAL_OCI=1 helm upgrade \
   --install trento-server oci://registry.suse.com/trento/trento-server \
   --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \
   --set trento-web.adminUser.password=ADMIN_PASSWORD
----
====
+
. To verify that the {trserver} installation was successful, open the URL of the {trweb} (`http://TRENTO_SERVER_HOSTNAME`) from a workstation on the {sap} administrator's LAN.

[[sec-trento-install-trentoserver-on-k3s]]
==== Installing {trserver} on K3s

> If you do not have a {k8s} cluster, or have one but do not want to use it for {trento}, {suse} Rancher's K3s provides an alternative. To deploy {trserver} on K3s, you need a small server or VM (see <<sec-trento-server-requirements>> for minimum requirements) and follow steps in <<pro-trento-manually-installing>>.

Reviewer: ```suggestion
If you do not have a {k8s} cluster, or you have one but you do not want to use it for {trento}, you can use {suse} Rancher's K3s as an alternative. To deploy {trserver} on K3s, you need a server or VM (see <<sec-trento-server-requirements>> for minimum requirements) and follow steps in <<pro-trento-manually-installing>>.
```
